src/parse.rs

```diff
@@ -1,8 +1,12 @@
-use std::error::Error;
+use std::{error::Error, path::Path};
 
 use serde::{Deserialize, Serialize};
 
-pub fn read_from_file<T: for<'de> Deserialize<'de>>(path: &str) -> Result<Vec<T>, Box<dyn Error>> {
+pub fn read_from_file<T, P>(path: P) -> Result<Vec<T>, csv::Error>
```

Comment:
Maybe also update `write_to_file`'s path type for consistency.

Reply:
Sorted